tools/oni_viewer_simple.cpp

```diff
@@ -206,5 +206,6 @@ main(int argc, char ** argv)
     v.run();
   }
 
+  delete(grabber);
```

Comment:
Use `unique_ptr` instead if pointer is needed. But it's not. Even here I don't see the need of a pointer.

Reply:
https://github.com/PointCloudLibrary/pcl/blob/6c1e94b6d06d66d0c08de4a18f82ff24a97332aa/tools/oni_viewer_simple.cpp#L195 
This wasn't working without a pointer. I will look into it again

Reply:
That's capture by value, hence needs a pointer. But you can capture by reference IFF you can guarantee that trigger will not outlast grabber.